packages/react-client/src/hooks/useDevices.ts

```diff
@@ -1,41 +1,62 @@
 import { useFishjamContext } from "./useFishjamContext";
-import type { Device, AudioDevice, DeviceState } from "../types";
+import type { Device, AudioDevice } from "../types";
 import { useVideoDeviceManager } from "./deviceManagers/useVideoDeviceManager";
 import { useAudioDeviceManager } from "./deviceManagers/useAudioDeviceManager";
 import { useCallback } from "react";
 import { getAvailableMedia, getCorrectedResult } from "../mediaInitializer";
-import type { Track } from "../state.types";
-
-function getDeviceProperties(currentTrack: Track | null, deviceState: DeviceState) {
-  const streamedTrackId = currentTrack?.trackId ?? null;
-  const streamedTrack = currentTrack?.track ?? null;
-  // todo: temporary fix, fix in FCE-450
-  const stream = currentTrack?.stream ?? deviceState.media?.stream ?? null;
-  const devices = deviceState.devices ?? [];
-  const activeDevice = deviceState.media?.deviceInfo ?? null;
-
-  return { streamedTrack, streamedTrackId, stream, devices, activeDevice };
-}
+import { useProcessedPreviewStream } from "./useProcessedPreviewStream";
 
 export function useCamera(): Device {
-  const {
-    videoTrackManager: { currentTrack, ...trackManager },
-  } = useFishjamContext();
-
+  const { videoTrackManager } = useFishjamContext();
   const { deviceState } = useVideoDeviceManager();
+  const { currentTrack, ...trackManager } = videoTrackManager;
+
+  const processedPreviewStream = useProcessedPreviewStream(videoTrackManager, deviceState.media?.track);
 
-  return { ...trackManager, ...getDeviceProperties(currentTrack, deviceState) };
+  const stream = currentTrack?.stream ?? processedPreviewStream ?? deviceState.media?.stream ?? null;
```

Comment:
Do we need to check `currentTrack?.stream`? 

```suggestion
  const stream = processedPreviewStream ?? deviceState.media?.stream ?? null;
```